src/indicator/gpg.ts

```diff
@@ -9,6 +8,70 @@ export interface GpgKeyInfo {
     keygrip: string;
 }
 
+/**
+ * Get the path of socket file for communication with GPG agent.
+ *
+ * @returns The path of desired GPG agent socket.
+ */
+async function getSocketPath(): Promise<string> {
+    // TODO: Consider supporting other socket files rather than the default one.
+    const outputs = await process.textSpawn('gpgconf', ['--list-dir', 'agent-socket'], "");
+
+    return outputs.trim();
+}
+
+/**
+ * Sign the given hash string with the specified GPG key.
+ *
+ * @param logger - The logger object for debugging logs.
+ * @param socketPath - The path of socket file to communicated with GPG agent.
+ * @param keygrip - The keygrip of the GPG key for the signing operation.
+ * @param passphrase - The passphrase of the key.
+ * @param sha1Hash - The hash string to be signed.
+ */
+async function sign(logger: assuan.Logger, socketPath: string, keygrip: string, passphrase: string, sha1Hash: string): Promise<void> {
+    let response: assuan.Response;
+    const agent = new assuan.AssuanClient(logger, socketPath);
+    await agent.initialize();
+    response = await agent.receiveResponse();
+    response.checkType(assuan.ResponseType.ok);
+
+    await agent.sendRequest(assuan.Request.fromCommand(new assuan.RequestCommand('OPTION', 'pinentry-mode loopback')));
+    response = await agent.receiveResponse();
+    response.checkType(assuan.ResponseType.ok);
+
+    await agent.sendRequest(assuan.Request.fromCommand(new assuan.RequestCommand('SIGKEY', keygrip)));
+    response = await agent.receiveResponse();
+    response.checkType(assuan.ResponseType.ok);
+
+    await agent.sendRequest(assuan.Request.fromCommand(new assuan.RequestCommand('SETHASH', `--hash=sha1 ${sha1Hash}`)));
+    response = await agent.receiveResponse();
+    response.checkType(assuan.ResponseType.ok);
+
+    await agent.sendRequest(assuan.Request.fromCommand(new assuan.RequestCommand('PKSIGN')));
+    response = await agent.receiveResponse();
+    let type = response.getType();
+    if (type === assuan.ResponseType.rawData) { // Key is already unlocked
```

Comment:
I prefer not using inline comments
```suggestion
    
    // Key is already unlocked
    if (type === assuan.ResponseType.rawData) { 
```